### PR TITLE
Focus API

### PR DESCRIPTION
**Pull request by kuzmany**
API support to https://github.com/mautic/mautic/pull/3460

### CODE REVIEW

**Comment by escopecz** [comment]
Could you create also the tests for this? We have all API endpoints covered with tests which test the library as well as Mautic side. See the example tests at the tests/Api dir. Here is a short tutorial: https://github.com/mautic/api-library#unit-tests

**Comment by kuzmany** [comment]
Coming soon :)

**Comment by kuzmany** [comment]
@escopecz 

I commit tests for Focus API, but I am little confused. I guess there is problem with entityNameOne and entityNameMulti.  Can you test it?

Also this PR contain fix for test with  BasicAuth   -  https://github.com/mautic/api-library/pull/93/commits/8b4d951983f8c39bf66ac14b033662e385fd3e8f

**Comment by kuzmany** [comment]
@escopecz  tests already worked.  PR is done.

**Comment by luizeof** [comment]
hi @escopecz when this will be merged?

**Comment by escopecz** [comment]
@luizeof thanks for the question!

It doesn't make sense to merge this until Mautic will support these API endpoints. So https://github.com/mautic/mautic/pull/3460 must be merged and released first. Why that PR was not merged already you may ask? Because it wasn't tested by at least 2 devs. This is where you could help to speed it up.

File: tests/Api/FocusTest.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    {
+        $this->api = $this->getContext('focus');
+        $this->testPayload = array(
+            'name' => 'test',
+            'type' => 'notice',
+            'website' => 'http://',
+            'style' => 'bar',
+            'htmlMode' => 1,
+            'html' => '<div><strong style="color:red">html mode enabled</strong></div>',
+            'css' => '.mf-bar-collapser {border-radius: 0 !important}',
```
Comment (escopecz) [inline tests/Api/FocusTest.php]: @kuzmany While running the final tests before the API Library 2.9.0 release, the `Mautic\Tests\Api\FocusTest::testBatchEndpoints` test fails on `The ["focus" => "css"] doesn't exist in the response`. When I look at the Focus entity class, it does not have the `css` property. Why is it here?

Comment (escopecz) [inline tests/Api/FocusTest.php]: Ok. I'll remove the css property from the test payload then.

**Pull request merged**